File: HamiltonCourtFX.Emporos.ITInventory.Common/Entities/Employee.cs
using System.ComponentModel.DataAnnotations;

namespace HamiltonCourtFX.Emporos.ITInventory.Common.Entities
{
    [Serializable]
    public class Employee
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Email { get; set; }

        public List<Device> Devices { get; set; }

        public Employee() 
        {
            Devices = new List<Device>();
        }
    }
}

File: hamiltoncourtfx.emporos.itinventory.client/src/app/models/employee.ts
import { DeviceInterface } from "../interfaces/device.interface";
import { EmployeeInterface } from "../interfaces/employee.interface";

export class EmployeeModel implements EmployeeInterface {
  constructor(
    public id: number,
    public name: string,
    public email: string,
    public devices: DeviceInterface[]
  ) { }

} 
